bulletroboy/exoforce.py

```diff
@@ -286,3 +294,122 @@ def get_motors(self):
 
 	def get_tendons(self):
 		return [ muscle.tendon for muscle in self.muscle_units ]
+
+
+class Movements():
+	def __init__(self, body_id):
+		"""
+		This class defines 2 types of movements:
+		-  Re-definable inverse kinematic ones (one_end_effector(), multiple_end_effectors())
+		-  Harcoded simple ones (simple_move() --> 4 predefined movements)
+		"""
+		self.body_id = body_id
+		self.op = Operator(self.body_id)
+
+
+	def get_EF(self, link_name):
+                freeJoints = []
+                numJoints = p.getNumJoints(self.body_id)
+
+                for i in range(numJoints):
+                    info = p.getJointInfo(self.body_id,i)
+                    if info[2] == p.JOINT_REVOLUTE:
+                        freeJoints.append(i)
+                    if info[12] == link_name:
+                            endEffectorId = i
+
+                return endEffectorId, freeJoints
+
+
+	def apply_chest_and_neck_constraints(self):
+		p.createConstraint(self.body_id, self.op.get_link_index('human/spine_1'), -1,
+				      self.op.get_link_index('human/spine_2'), p.JOINT_FIXED,
+				      [0, 0, 0], [0, 0, 0], [0, 0, 1])
+
+		p.createConstraint(self.body_id, self.op.get_link_index('human/neck'), -1, 					      self.op.get_link_index('human/spine_2'), p.JOINT_FIXED,
+				      [0, 0, 0], [0, 0, 0], [0, 0, 1])
+
+
+	def one_end_effector(self, link_name, pos, maxIter, chest_constraint):
+		error = False
+		endEffectorId, freeJoints = self.get_EF(link_name)
+		if chest_constraint == True:
+		    self.apply_chest_and_neck_constraints()
+		iter = 0
+		try:
+		    while(iter <= maxIter):
+		        jointPoses = p.calculateInverseKinematics(self.body_id, endEffectorId, pos)
+		        for i in range(len(freeJoints)):
+          		    p.resetJointState(self.body_id, freeJoints[i], jointPoses[i])
+		        iter = iter + 1
+		except SystemError:
+		    print(colored('\nSystemError:', 'red'))
+		    print(colored('COULD NOT UPDATE JOINT STATES: check if end effector position is realistic!!!\n\n', 'red'))
+		    error = True
+
+		return error
+
+
+	def two_end_effectors(self, positions, maxIter, chest_constraint):
+		error = False
+		_, freeJoints = self.get_EF(b'human/left_hand')
+		endEffectorIds = [self.op.get_link_index('human/left_hand'),
+                                  self.op.get_link_index('human/right_hand')]
+		
+		if chest_constraint == True:
+		    self.apply_chest_and_neck_constraints()
+		iter = 0
+
+		try:
+		    while(iter <= maxIter):
+		        jointPoses = p.calculateInverseKinematics2(self.body_id, endEffectorIds, positions)
+		        for i in range(len(freeJoints)):
+		            p.resetJointState(self.body_id, freeJoints[i], jointPoses[i])
+		        iter = iter + 1
+		except SystemError:
+		    print(colored('\nSystemError:', 'red'))
+		    print(colored('THE COMBINATION OF POSITIONS USED AS INPUT IN two_end_effectors() IS NOT ACCEPTABLE!!!\n\n', 'red'))
+		    error = True
+
+
+		return error
+
+
+	def simple_move(self, case):
```

Comment:
- [ ] define constants for each case and use integers instead of strings. one should be able to call this using `operator.simple_move(operator.SPINE_SWING)` (of course the syntax might change depending on your implementation, but you get the idea :) )

Reply:
I have a question about this. Should I put simple_move inside the operator class or do I do what you did but with the movements class instead of the operator one?

mv=Movements(operator)
mv.simple_move(mv.SPINE_SWING)

Reply:
ideally it's 
```python
from movement.primitives import SPINE_SWING
operator.move(SPINE_SWING)
``` 
what do you think? can we get there?

Reply:
done :)